src/services/platform.service.js

```diff
@@ -72,33 +72,38 @@ export const getBaseProjectEnvironment = (
   };
 };
 
-window.childProcess = childProcess;
-
 // HACK: With electron-builder, we're having some issues on mac finding Node.
 // This is because for some reason, the PATH is not updated properly :(
 // 'fix-path' is supposed to do this for us, but it doesn't work, for unknown
 // reasons.
 export const initializePath = () => {
-  childProcess.exec('which node', { env: window.process.env }, (_, version) => {
-    if (!version && isMac) {
-      // For users with a standard Node installation, node will be in
-      // /usr/local/bin
-      // For users using NVM, the path to Node will be added to `.bashrc`.
-      // Add both to the PATH.
-      try {
-        childProcess.exec(
-          'source ~/.bashrc && echo $PATH',
-          (err, updatedPath) => {
-            if (updatedPath) {
-              window.process.env.PATH = `/usr/local/bin:${updatedPath}`;
-            }
+  return new Promise((resolve, reject) => {
+    childProcess.exec(
+      'which node',
+      { env: window.process.env },
+      (_, version) => {
+        if (!version && isMac) {
+          // For users with a standard Node installation, node will be in
+          // /usr/local/bin
+          // For users using NVM, the path to Node will be added to `.bashrc`.
+          // Add both to the PATH.
+          try {
+            childProcess.exec(
```

Comment:
This is not going to throw since there is a callback, or am I missing something?